src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/SpgProducerEeeCsvDataTests.cs

```diff
@@ -64,6 +65,192 @@ public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligat
+                var maxCategoryId = categories.Max(x => x.Key);
+                int quarter3Tonnage = 0;
+                int quarter4Tonnage = 0;
+
+                for (int i = 1; i <= maxCategoryId; i++)
+                {
+                    quarter3Tonnage = quarter3Tonnage + 10;
+                    quarter4Tonnage = quarter4Tonnage + 5;
+                    helper.CreateEeeOutputAmount(dataReturnVersion1, producer1.RegisteredProducer, b2bObligationType, i, quarter3Tonnage);
+                    helper.CreateEeeOutputAmount(dataReturnVersion2, producer1.RegisteredProducer, b2bObligationType, i, quarter4Tonnage);
+                }
+
+                helper.CreateEeeOutputAmount(dataReturnVersion1, producer2.RegisteredProducer, b2cObligationType, 2, 400);
+
+                db.Model.SaveChanges();
+
+                // Act
+                var results = await db.StoredProcedures.SpgProducerEeeCsvData(complianceYear, null, b2bObligationType, false, false);
+
+                //Assert
+                Assert.NotNull(results);
+
+                //Data return with obliation type B2B should not be there in the result.
+                ProducerEeeCsvData b2bProducer = results.Find(x => (x.PRN == prn2));
+                Assert.Null(b2bProducer);
+
+                ProducerEeeCsvData result = results[0];
+                Assert.Equal(approvalNumber, result.ApprovalNumber);
+                Assert.Equal(prn1, result.PRN);
+                Assert.Equal(10, result.Cat1Q3);
+                Assert.Equal(20, result.Cat2Q3);
+                Assert.Equal(30, result.Cat3Q3);
+                Assert.Equal(40, result.Cat4Q3);
+                Assert.Equal(50, result.Cat5Q3);
+                Assert.Equal(60, result.Cat6Q3);
+                Assert.Equal(70, result.Cat7Q3);
+                Assert.Equal(80, result.Cat8Q3);
+                Assert.Equal(90, result.Cat9Q3);
+                Assert.Equal(100, result.Cat10Q3);
+                Assert.Equal(110, result.Cat11Q3);
+                Assert.Equal(120, result.Cat12Q3);
+                Assert.Equal(130, result.Cat13Q3);
+                Assert.Equal(140, result.Cat14Q3);
+                Assert.Equal(150, result.Cat15Q3);
+
+                Assert.Equal(5, result.Cat1Q4);
+                Assert.Equal(10, result.Cat2Q4);
+                Assert.Equal(15, result.Cat3Q4);
+                Assert.Equal(20, result.Cat4Q4);
+                Assert.Equal(25, result.Cat5Q4);
+                Assert.Equal(30, result.Cat6Q4);
+                Assert.Equal(35, result.Cat7Q4);
+                Assert.Equal(40, result.Cat8Q4);
+                Assert.Equal(45, result.Cat9Q4);
+                Assert.Equal(50, result.Cat10Q4);
+                Assert.Equal(55, result.Cat11Q4);
+                Assert.Equal(60, result.Cat12Q4);
+                Assert.Equal(65, result.Cat13Q4);
+                Assert.Equal(70, result.Cat14Q4);
+                Assert.Equal(75, result.Cat15Q4);
+
```

Comment:
would it be worth doing both B2C and B2B in one test, but with different numbers so they're definitely the right ones?

Reply:
you might have a good reason for doing them separately, and that's fine if so, but I'm not sure what it is

Reply:
oh -- I understand this now -- it's adding lots of b2b/b2c data, then a bit for the other one, then checking only one type is there. I think that's worth a comment, as it's quite subtle. Is there a reason for not checking all four quarters in one test though?